examples/ml/housing-prices/README.md

```diff
@@ -0,0 +1,247 @@
+4. Generate a learning curve and performance metrics to estimate the quality of the model
+
+Outline of this Example: 
+
+- [Boston Housing Prices](#boston-housing-prices)
+  - [Housing prices datasets](#housing-prices-datasets)
+  - [Getting Started](#getting-started)
+  - [Python code](#python-code)
+    - [Data Analysis](#data-analysis)
+    - [Train a regression model](#train-a-regression-model)
+    - [Evaluate the model](#evaluate-the-model)
+  - [Pachyderm Pipeline](#pachyderm-pipeline)
+    - [TLDR; Just give me the code](#tldr-just-give-me-the-code)
+    - [Step 1: Create input data repository](#step-1-create-input-data-repository)
+    - [Step 2: Create the regression pipeline](#step-2-create-the-regression-pipeline)
+    - [Step 3: Add the housing dataset to the repo](#step-3-add-the-housing-dataset-to-the-repo)
+    - [Step 4: Download files once the pipeline has finished](#step-4-download-files-once-the-pipeline-has-finished)
+    - [Step 5: Update Dataset](#step-5-update-dataset)
+
+## Housing prices datasets
+
+The housing prices dataset used for this example is a reduced version of the original, focusing on a subset of the features. It contains 3 input features (RM, LSTST, and PTRATIO) and the output, or target (MEDV) that we are learning to predict.
+|Feature| Description|
+|---|---|
+|RM |       Average number of rooms per dwelling|
+|LSTAT |    Percent lower status of the population|
+|PTRATIO |  Pupil-teacher ratio by town|
+|MEDV |     Median value of owner-occupied homes in $1000's|
+
+Sample:
+|RM   |LSTAT|PTRATIO|MEDV|
+|-----|----|----|--------|
+|6.575|4.98|15.3|504000.0|
+|6.421|9.14|17.8|453600.0|
+|7.185|4.03|17.8|728700.0|
+|6.998|2.94|18.7|701400.0|
+
+## Getting Started
+
+This example requires a running Pachyderm deployment. You can deploy a cluster on [PacHub](hub.pachyderm.com) or deploy locally as described here:
+
+- [Pachyderm Getting Started](https://docs.pachyderm.com/latest/getting_started/)
+
+Once everything is up, we can check the setup by running:
+
+1. `kubectl get all` to ensure all the pods are up.
+2. `pachctl version` which will show both the `pachctl` and `pachd` versions.
+
+Next, clone this repo and follow the steps below.
+
+
+
+## Python code
+
+The `regression.py` Python file contains the machine learning code for the example. We will give a brief description of it here, but full knowledge of it is not required for the example. 
+
+```
+$ python regression.py --help
+
+usage: regression.py [-h] [--input INPUT] [--target-col TARGET_COL]
+                     [--output DIR]
+
+Structured data regression
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --input INPUT         csv file with all examples
+  --target-col TARGET_COL
+                        column with target values
+  --output DIR          output directory
+  ```
+
+The regression code performs three functions:
+
+1. Data analysis
+2. Train a regressor
+3. Evaluate the model
```

Comment:
is there a reason we have just one pipeline step that does all 3 of these? is it common for this type of model/process to do that all together or would these typically be broken up into 2 or 3 different pipeline steps?

Reply:
I kept them all in one pipeline to keep the example simple, but also was setting up for potentially doing multiple pipelines in the future. Any of these functions would need access to the data, so it would make sense to execute some of them in parallel pipelines, such as data analysis and training. But we would want training and eval to be in the same pipeline unless we separated out the test data into another repo.